### ISSUE TITLE
Does not handle Pandas 'NA' values

### ISSUE THREAD

**Issue by mangecoeur** [bloomberg/ipydatagrid#257]
Pandas provides it's own 'NA' object for cases where numpy NaN is not appropriate (e.g. string columns). This is not currently handled in `_data_to_json` (only pandas NaT values are handled).

**Comment by ibdafna** [comment]
Hello @mangecoeur 👋 and welcome to the `ipydatagrid` community!

Thank you for taking the time to file this issue and submit #258. Is there an edge case that the current serialisation function does not capture? I am basically trying to understand if there's a need to add that additional check. As you correctly stated, adding `pd.isna("Hello")`, for example, will result in the predicate evaluating to `True`, which in turn will break rendering strings in the grid.

**Comment by mangecoeur** [comment]
Hi @ibdafna, thanks for the reply. Yes there is a simple case where this breaks:

```python
import pandas as pd
from ipydatagrid import DataGrid
df = pd.DataFrame({'text_col' : ['a', 'b', pd.NA, 'd']})
DataGrid(df)
```

Gives:

`ValueError: Can't clean for JSON: <NA>`

Although I haven't managed to run tests yet to see if the additional check solves this, I'm pretty sure it should. There's not a huge imperative for this, in a string column can set `None` instead of `NA` to identify missing values, but i think for completeness it makes sense that pd.NA values should work (also I think pandas is introducing more elaborate NA handling in e.g. integer and string columns so this might crop up more in the future).

**Comment by ibdafna** [comment]
@mangecoeur Okay, thank you - I understand this better now 😸 . I will comment in the PR!

**Comment by ibdafna** [comment]
Closed via #258 . Thank you @mangecoeur !

**Comment by JohnOmernik** [comment]
Just a FYI, I think this also applies to Decimal as I am getting "Can't clean for JSON: Decimal('0.00')"
